### ISSUE TITLE
Numbers are treated as different types when run from editor than from release builds

### ISSUE THREAD

**Issue by shelbyd** [godotengine/godot#56609]
### Godot version

3.2.3.stable

### System information

Windows 10

### Issue description

Exporting a project removes type information, causing floats to be treated as ints.

It is a severe concern that something can work in the editor version and not in an exported version. This would require me to re-check every change I make with the exported version.

### Steps to reproduce

Given the following GDScript:

```gd
export(float) var something = 3
var other

func _ready():
  other = something / 8
```

When run from the editor, `other` will be `0.375`. When run from an exported project, `other` will be 0.

**Comment by Calinou** [comment]
> 3.2.3.stable

[Godot 3.2.x is not supported anymore](https://docs.godotengine.org/en/stable/about/release_policy.html#release-support-timeline), so please upgrade to [3.4.2](https://godotengine.org/article/maintenance-release-godot-3-4-2) and check if the issue still occurs there.

**Comment by KoBeWi** [comment]
Sounds similar to #42124
What if you declare the variable as `export var something: float = 3` instead?

**Comment by shelbyd** [comment]
This behavior is the same in 3.4.2-stable.

If I add the type as an annotation as `export var something: float = 3`, then `other` is `0.375` in both the editor and exported project, as expected.

**Comment by akien-mga** [comment]
Confirmed in 3.5 beta 1, and I've checked and it wasn't fixed by #57851.

**Comment by akien-mga** [comment]
Duplicate of #39215.
